packages/lab/src/carousel/Carousel.tsx

```diff
@@ -0,0 +1,191 @@
+  const [selectedTransition, setSelectedTransition] = useState<
+    LayoutAnimationTransition | undefined
+  >(undefined);
+
+  const handleSlideSelection = (
+    sliderIndex: number,
+    transition?: LayoutAnimationTransition
+  ) => {
+    const newTransition = transition
+      ? transition
+      : selectedSlide < sliderIndex
+      ? "increase"
+      : "decrease";
+    setSelectedSlide(sliderIndex);
+    setSelectedTransition(newTransition);
+  };
+  return [selectedTransition, selectedSlide, handleSlideSelection];
+};
+export const Carousel = forwardRef<HTMLDivElement, CarouselProps>(
+  function Carousel(
+    {
+      activeIndex,
+      animation = "slide",
+      animationTimeout = 800,
+      carouselDescription,
+      children,
+      className,
+      compact,
+      direction,
+      id: idProp,
+      ...rest
+    },
+    ref
+  ) {
+    const id = useId(idProp);
+    const slidesCount = Children.count(children);
+
+    const [selectedTransition, selectedSlide, handleSlideSelection] =
+      useSlideSelection(activeIndex);
+
+    const moveSlide = (direction: SlideDirections) => {
+      const moveLeft =
+        selectedSlide === 0 ? slidesCount - 1 : selectedSlide - 1;
+      const moveRight =
+        selectedSlide === slidesCount - 1 ? 0 : selectedSlide + 1;
+      const newSelection = direction === "left" ? moveLeft : moveRight;
+      const newTransition = direction === "left" ? "decrease" : "increase";
+      handleSlideSelection(newSelection, newTransition);
+    };
+
+    const handleRadioChange: ChangeEventHandler<HTMLInputElement> = ({
+      target: { value },
+    }) => {
+      handleSlideSelection(Number(value));
+    };
+
+    useEffect(() => {
+      if (process.env.NODE_ENV !== "production") {
+        const validNumberOfChildren = slidesCount > 1;
+        warning(
+          validNumberOfChildren,
+          "Carousel component requires more than one children to render. At least two elements should be provided."
+        );
+      }
+    }, [children]);
+
+    return (
+      <div
+        aria-label={carouselDescription}
+        aria-roledescription="carousel"
+        id={id}
+        role="region"
+        ref={ref}
+        {...rest}
+        className={cx(
+          withBaseName(),
+          compact && withBaseName("compact"),
+          className
+        )}
+      >
+        <Button
+          variant="secondary"
+          className={withBaseName("prev-button")}
+          onClick={() => moveSlide("left")}
+        >
+          <ChevronLeftIcon size="medium" />
+        </Button>
+        <DeckLayout
+          activeIndex={selectedSlide}
+          animation={animation}
+          transition={selectedTransition}
+          className={withBaseName("slider")}
+          direction={direction}
+        >
+          {children}
+        </DeckLayout>
+        <Button
+          variant="secondary"
+          className={withBaseName("next-button")}
+          onClick={() => moveSlide("right")}
```

Comment:
When clicking the button, main slides is flicking before starting the animation

Reply:
Should be fixed now